src/gsCore/gsRationalBasis.h

```diff
@@ -136,14 +136,17 @@ class gsRationalBasis : public gsBasis< typename SrcT::Scalar_t >
 
     index_t size() const { return m_src->size(); }
 
-    int size(int const& k) const{ return m_src->size(k); }
+    index_t size(index_t const& k) const{ return m_src->size(k); }
```

Comment:
Maybe use the types from here:
https://github.com/gismo/gismo/blob/3fa0f13f5aadd03170c8229e41640ab084d11043/src/gsTensor/gsTensorBasis.h#L437C7-L437C65